Source/ARTRestChannel.m

```diff
@@ -247,9 +247,6 @@ - (void)internalPostMessages:(id)data callback:(void (^)(ARTErrorInfo *__art_nul
             callback([ARTErrorInfo createWithCode:ARTStateMismatchedClientId message:@"attempted to publish message with an invalid clientId"]);
             return;
         }
-        else {
```

Comment:
Could you explain somewhere in this PR, or against the original issue even, why this change was necessary and why it fixes the issue?

Reply:
I think the removed code existed because of a misinterpretation of the spec. The library should not change the clientId. The clientId is explicitly configured with the `ClientOptions` or implicitly through Token Auth.